inc/parser/xhtml.php

```diff
@@ -59,16 +59,24 @@ class Doku_Renderer_xhtml extends Doku_Renderer {
     /**
      * Register a new edit section range
      *
-     * @param string $type   The section type identifier
-     * @param string $title  The section title
      * @param int    $start  The byte position for the edit start
+     * @param array  $data   Assoziative array with section data:
+     *                       Key 'name': the section name/title
+     *                       Key 'target': the target for the section edit,
+     *                                     e.g. 'section' or 'table'
+     *                       Key 'hid': header id
+     *                       Key 'codeblockOffset': actual code block index
+     *                       Key 'start': reserved, do not use
+     *                       Key 'range': reserved, do not use
```

Comment:
I don't think these should be marked as reserved. Actually they are required, aren't they?

Reply:
I also wonder about having `start` and `range`. Shouldn't we have `start` and `end` instead?

Reply:
The ```$start``` value is passed to ```startSectionEdit()``` as an extra parameter. The ```$end``` value is given to ```finishSectionEdit()``` as an extra parameter. In ```finishSectionEdit()``` finally the saved ```data['start']``` value and given ```$end``` value are used to build the ```range```. The ```range``` is the important field that is sent on clicking on the section edit button. If that is not found then the section edit will not work. So, start and end are only temporary values and I thought it is better to put the calculation of the range into ```startSectionEdit()``` and ```finishSectionEdit()``` instead of having the caller calculating and assigning the range to the array. That means the keys start and range will always be overwritten. That's why I thought it would be good to mention them as reserved.

Reply:
I see. Maybe we can get the gist of that explanation into the doc block? Like:

```
'start': set in startSectionEdit(), do not set yourself
'range': calculated from 'start' and $key in finishSectionEdit(), do not set yourself
```

Something like that. I'm open for improvements on that.

Reply:
Sounds good, have used it.